### PR TITLE
Features/gat 86 Reject/Approve a dataset

### PR DESCRIPTION
**Pull request by sikenyonhdruk**

### CODE REVIEW

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.js [hunk 1]
```diff
@@ -176,8 +186,42 @@ const AccountDataset = props => {
...
+								Make a decision
+							</Button>
+						</OverlayTrigger>
 					</div>
 				</ActionBar>
+				<AccountDatasetApproveModal
+					id={currentDataset._id}
+					open={showApproveDatasetModal}
+					closed={() => setState({ showApproveDatasetModal: false })}
+					goToNext={() => goToNext()}
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.js:217]: goToNext={goToNext}

const goToNext = React.useCallback(() ={}, [])

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.js [hunk 2]
```diff
@@ -176,8 +186,42 @@ const AccountDataset = props => {
...
+					showGoToNext={showNext} />
+				<AccountDatasetRejectModal
+					id={currentDataset._id}
+					open={showRejectDatasetModal}
+					closed={() => setState({ showRejectDatasetModal: false })}
+					goToNext={() => goToNext()}
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.js:223]: goToNext={goToNext}

const goToNext = React.useCallback(() ={}, [])

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.test.js [hunk 3]
```diff
@@ -86,6 +85,75 @@ describe('Given the AccountDataset component', () => {
...
+				});
+
+				it('Then loads the Approve dataset modal', () => {
+					const { queryByText } = wrapper;
+					expect(queryByText('Approve this version of this dataset metadata')).toBeTruthy();
+				});
+
+				it('Then does not load the Reject dataset modal', () => {
+					const { queryByText, debug } = wrapper;
+					debug();
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.test.js:151]: Assume this is left over from modal problem tests

File: src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+				<CloseButtonSvg className='account-dataset-decision-modal-close' onClick={closed} />
+				<div className='account-dataset-decision-header--wrap'>
+					<h4 className='black-20'>Approve this version of this dataset metadata</h4>
+				</div>
+			</div>
+
+			<div className='account-dataset-decision-body'>
+				<div className='account-dataset-decision-body--wrap'>
+					<p>If you made any amendments to their application, such as fixing a type, please let the editor know using the description below.</p>
+					<label className='black-14'>Description (optional)</label>
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.js:58]: forHtml="applicationStatusDesc"

File: src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+			<div className='account-dataset-decision-header'>
+				<CloseButtonSvg className='account-dataset-decision-modal-close' onClick={closed} />
+				<div className='account-dataset-decision-header--wrap'>
+					<h4 className='black-20'>Approve this version of this dataset metadata</h4>
+				</div>
+			</div>
+
+			<div className='account-dataset-decision-body'>
+				<div className='account-dataset-decision-body--wrap'>
+					<p>If you made any amendments to their application, such as fixing a type, please let the editor know using the description below.</p>
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.js:57]: Translations, tests _should_ work if you do this:

```
wrapper = render(<ApproveModal {...props} />, {
    wrapper: Providers,
});
```

Providers is global.Providers which gives you:

```
		<I18nextProvider i18n={i18n}>
			<Suspense fallback='Loading'>
				<ThemeProvider theme={theme}>
					<AuthProvider value={{ userState: mockUser.data }}>
						<QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
					</AuthProvider>
				</ThemeProvider>
			</Suspense>
		</I18nextProvider>
```

File: src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.test.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        goToNext: jest.fn(),
+        showGoToNext: true
+    };
+
+    describe('When it is rendered', () => {
+        let wrapper;
+
+        beforeAll(() => {
+            containerDiv = createPortalContainer();
+            wrapper = render(
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.test.js:32]: As above, you can use this (and strip out QueryClient):

```
wrapper = render(<ApproveModal {...props} />, {
    wrapper: Providers,
});
```

A few places below as well, I won't repeat comment!

File: src/pages/dashboard/Components/AccountDataset/AccountDatasetRejectModal.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+				<CloseButtonSvg data-testid="close-modal" className='account-dataset-decision-modal-close' onClick={closed} />
+				<div className='account-dataset-decision-header--wrap'>
+					<h4 className='black-20'>Reject this version of this dataset metadata</h4>
+				</div>
+			</div>
+
+			<div className='account-dataset-decision-body'>
+				<div className='account-dataset-decision-body--wrap'>
+					<p>Let the editor know why this submission is being rejected. They will be able to create a new version and make a new submission.</p>
+					<label className='black-14'>Description</label>
```
Comment (ThinkingMonkeyLtd) [inline src/pages/dashboard/Components/AccountDataset/AccountDatasetRejectModal.js:58]: forHtml="applicationStatusDesc"

File: src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.test.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                )
+            });
+            it('Then the Approve and go to next button should be disabled', () => {
+                const { getByText } = wrapper;
+                const approveAndGoToNextButton = getByText('Approve and go to next');
+                expect(approveAndGoToNextButton).toHaveAttribute('disabled');
+            });
+        });
+    });
+});
```
Comment (kandaj) [inline src/pages/dashboard/Components/AccountDataset/AccountDatasetApproveModal.test.js:74]: test cases for Approve and go to next click  are missing.

File: src/pages/dashboard/Components/AccountDataset/AccountDatasetRejectModal.test.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            it('Should render an error message', async () => {
+                const { getByTestId, findByText } = wrapper;
+                const rejectButton = getByTestId('reject-button');
+                fireEvent.click(rejectButton);
+
+                expect(await findByText('Description cannot be empty')).toBeTruthy();
+            });
+        });
+    });
+});
```
Comment (kandaj) [inline src/pages/dashboard/Components/AccountDataset/AccountDatasetRejectModal.test.js:84]: test cases for Reject and go to next click  are missing.

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.js [hunk 10]
```diff
@@ -176,8 +186,42 @@ const AccountDataset = props => {
...
+							<Button variant='outline-secondary'>
+								Make a decision
+							</Button>
+						</OverlayTrigger>
 					</div>
 				</ActionBar>
+				<AccountDatasetApproveModal
+					id={currentDataset._id}
+					open={showApproveDatasetModal}
+					closed={() => setState({ showApproveDatasetModal: false })}
```
Comment (kandaj) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.js:216]: you might need to setState for showPrevious and showNext

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.js [hunk 11]
```diff
@@ -176,8 +186,42 @@ const AccountDataset = props => {
...
+							key='top' 
+							placement='top'
+							overlay={
+								<Popover id='make-a-decision-popover'>
+									<Popover.Title>
+										Make a decision
+									</Popover.Title>
+									<Popover.Content>
+										<Button variant="link" onClick={() =>
+												setState({ showApproveDatasetModal: true })}>Approve</Button>
```
Comment (kandaj) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.js:200]: you might need to setState for showPrevious and showNext

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.js [hunk 12]
```diff
@@ -176,8 +186,42 @@ const AccountDataset = props => {
...
+										<Button variant="link" onClick={() =>
+												setState({ showRejectDatasetModal: true })}>Reject</Button>
+									</Popover.Content>
```
Comment (kandaj) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.js:203]: you might need to setState for showPrevious and showNext

**Review by ThinkingMonkeyLtd — APPROVED** [review]

File: src/pages/dashboard/Components/AccountDataset/AccountDataset.js [hunk 13]
```diff
@@ -111,6 +111,7 @@ const AccountDataset = props => {
 					type: 'dataset',
 				});
 			}
+			console.log(state.showNext);
```
Comment (kandaj) [inline src/pages/dashboard/Components/AccountDataset/AccountDataset.js:114]: remove console.log

Comment (kandaj) [reply src/pages/dashboard/Components/AccountDataset/AccountDataset.js:114]: Approve and Go to Next is still disabled. Same For Reject and Go to Next.

**Review by kandaj — APPROVED** [review]

**Pull request merged**